plugins/net_plugin/include/eosio/net_plugin/net_plugin.hpp

```diff
@@ -39,9 +39,48 @@ namespace eosio {
         std::optional<connection_status>  status( const string& endpoint )const;
         vector<connection_status>         connections()const;
 
+        struct p2p_per_connection_metrics {
+            struct connection_metric {
+               boost::asio::ip::address_v6::bytes_type address;
+               unsigned short port;
+               bool accepting_blocks;
+               uint32_t last_received_block;
+               uint32_t first_available_block;
+               uint32_t last_available_block;
+               size_t unique_first_block_count;
+               uint64_t latency;
+               size_t bytes_received;
+               time_t last_bytes_received;
```

Comment:
Instead of `time_t` probably better to use `std::chrono::nanoseconds`. See https://github.com/AntelopeIO/leap/pull/1321#discussion_r1235770984

Reply:
Converted to `chrono::nanoseconds`.